doc/README.md

```diff
@@ -99,22 +99,36 @@ Metamorph is designed to be horizontally scalable. As a result, the metamorphs d
+## Extended Format (EF) and Background Evaluation Extended Format (BEEF)
+
+For optimal performance, ARC uses custom formats for transactions.
+
+The first format is called the extended format, and is a superset of the raw transaction format. The extended format includes the satoshis and scriptPubKey for each input, which makes it possible for ARC to validate the transaction without having to download the parent transactions. In most cases the sender already has all the information from the parent transaction, as this is needed to sign the transaction.
 
 The only check that cannot be done on a transaction in the extended format is the check for double spends. This can only be done by downloading the parent transactions, or by querying a utxo store. A robust utxo store is still in development and will be added to ARC when it is ready. At this moment, the utxo check is performed in the Bitcoin node when a transaction is sent to the network.
 
-### Extended format efficiency
+The second format is called Background Evaluation Extended Format, or BEEF, in short. BEEF was created to enable and facilitate Simple Payment Verification (SPV) when sending transactions between peers and to allow for validation of transactions which inputs may not yet be mined. Although this format is mainly used in peer-to-peer transactions, a peer ultimately has to submit the transaction to the nodes, and to help with that, Arc not only accepts that format, but also performs the SPV.
+
+BEEF includes the transaction which constitutes the payment, as well as transactions whose outputs are used as inputs to the payment transaction (parent transactions) with their corresponding Merkle paths in the form of [BUMP](https://bsv.brc.dev/transactions/0074). In cases where one or more inputs are not yet mined, each ancestral transaction is included, stepping back through the Transaction DAG until every input has a corresponding parent transaction with a Merkle path.
+
+Arc validates each unmined transaction in BEEF in the same way it does an Extended Transaction. For each transaction with Merkle path (BUMP), the Merkle root is calculated and verified against [BlockTx](#BlockTx), which will prove that the Merkle roots provided are all part of block headers within the longest chain.
+
+Storing all block headers is a very network-heavy process, therefore Arc stores information only about the most recent blocks and verifies Merkle roots against them. If an older Merkle root is included in BEEF, and everything else during the transaction validation checks out, the transaction is left to be validated by the nodes. The premise for that is twofold:
+1. transactions are usually made from fresh UTXOs,
+2. full SPV should also be performed in peer-to-peer transaction process (e.g. between SPV wallets), where at least one of the peers has an independent source of Block Headers indexed by Merkle Root.
```

Comment:
I was hesitant to write this section. Let me know what you think, we can also remove it.

Reply:
Looks good to me